src/client/components/table-of-contents/table-of-contents.vue

```diff
@@ -14,5 +18,148 @@
 <script>
 export default {
   props: ['items'],
+  data () {
+    return {
+      activeItem: this.items[0].slug,
+      tableOfContentsIsOpen: false,
+    }
+  },
+  
+  mounted () {
+    window.addEventListener('scroll', this.updateActiveItem, { passive: true })
+  },
+
+  beforeDestroy () {
+    window.removeEventListener('scroll', this.updateActiveItem)
+  },
+
+  methods: {
+    updateActiveItem () {
+      const sectionsLenght = this.items.length
+
+      for (var i = 0; i < sectionsLenght; i++) {
+        const section = document.getElementById(this.items[i].slug)
+
+        if (this.isBelowScroll(section)) {
+          this.activeItem = section.id
+        }
+      }
+    },
+
+    isBelowScroll (el) {
+      const { top, height } = el.getBoundingClientRect()
+      
+      return top < 200 && ((top + height) > 0)
```

Comment:
can you move `200` to a variable with a descriptive name, so other developers will know what it means?